src/u2f.c

```diff
@@ -101,6 +106,82 @@ typedef struct __attribute__((__packed__)) {
     uint8_t challenge[U2F_NONCE_LENGTH];
 } U2F_AUTHENTICATE_SIG_STR;
 
+typedef enum {
+    U2F_STATE_IDLE,
+    U2F_STATE_AUTH,
+    U2F_STATE_REGISTER,
+} u2f_state_t;
+
+/**
+ * Holds state data for the U2F_STATE_AUTH workflow.
+ */
+typedef struct {
+    /** Part of the auth we're currently processing. */
+    enum {
+        U2F_STATE_AUTH_REQUESTED,
+        U2F_STATE_AUTH_COMPLETED,
+        U2F_STATE_AUTH_ERROR,
+        U2F_STATE_AUTH_HIJACK
```

Comment:
especially this, not sure what hijack means here. also consider renaming it, because "u2f hijack" is kind of a reserved word for smuggling data over u2f when a webwallet (like myetherwallet) bridges communication to the hardware wallet.

Reply:
Problem is: I have no idea what this is about as well! This has been refactored from the same "hijack" code in `master`, which seems to be a very convoluted no-op. It's been left in the code not to break the existing code flow. See https://github.com/digitalbitbox/bitbox02-firmware/blob/544a6690de4b7f8fe220852849ada0cd87a500aa/src/u2f.c#L441 and https://github.com/digitalbitbox/bitbox02-firmware/blob/544a6690de4b7f8fe220852849ada0cd87a500aa/src/u2f.c#L58